app/controllers/home_controller.rb

```diff
@@ -4,17 +4,26 @@ def index
     home_path_for_user
   end
 
-  # FIX: Temporary controller, remove me when no longer required
   def quality
     @scores = QualityScore.all.order(median: :desc, highest: :desc)
   end
 
+  def dashboard
+    @datasets_count = Dataset.count
+    @datafiles_count = Link.count
+    @publishers_count = Organisation.count
+    @published_datasets_count = Dataset.published.count
+    @draft_datasets_count = Dataset.draft.count
+    @unpublished_datasets_count = Dataset.left_outer_joins(:datafiles).where(links: { id: nil } ).count # i.e. datasets with no datafiles
```

Comment:
Perhaps move to a class method called 'datasets_with_no_datafiles' ?

Reply:
good idea